packages/asset/contracts/Catalyst.sol

```diff
@@ -161,6 +189,14 @@ contract Catalyst is
         uint256 tokenId,
         string memory metadataHash
     ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        require(
+            tokenId > 0 && tokenId < tokenCount,
+            "Catalyst: invalid token id"
+        );
```

Comment:
you have this require in multiple places, can we regroup in one modifier maybe ? (line 114)